src/main/scala/com/webtrends/harness/component/akkahttp/AkkaHttpBase.scala

```diff
@@ -63,11 +64,25 @@ trait AkkaHttpBase extends PathDirectives with MethodDirectives with AccessLog w
 
   // The AkkaHttpCORS trait is provided to enable CORS if desired
   protected val corsEnabled: Boolean = false
+  def allowedOrigins: Seq[HttpOrigin] = Seq()
 
   // Allows setting custom CORS settings by endpoint.
   // Default behavior allows all origins, allows credentials, and allows all methods defined in this Command for the path
   protected def corsSettingsByPath(path: String): CorsSettings = {
-    AkkaHttpCORS.corsSettings(immutable.Seq(method))
+    AkkaHttpCORS.corsSettings(immutable.Seq(method), allowedOrigins)
+  }
+
+  def corsRejectionHandler: RejectionHandler = {
+    RejectionHandler
+      .newBuilder()
+      .handleAll[javadsl.CorsRejection] { rejections =>
```

Comment:
Same question as above - why javadsl?

Reply:
This method is a replica of the corsRejectionHandler in "akka-http-cors - [scaladsl/CorsDirectives.scala](https://github.com/lomigmegard/akka-http-cors/blob/master/akka-http-cors/src/main/scala/ch/megard/akka/http/cors/scaladsl/CorsDirectives.scala)" 
We have this method here for 2 reasons -  To be able to return "Forbidden" error instead of BadRequest.
And log the failure in the access log.